File: 39.rb
require 'benchmark'

solve = Proc.new do
  perimeters = Hash.new(0)

  (1..500).each do |a|
    (1..500).each do |b|
      c = Math.sqrt(a**2 + b**2)
      p = a + b + c
      next if !(c.to_i == c) || p > 1000
      perimeters[p.to_i] += 1
    end
  end

  perimeters.max_by { |_, v| v }.inspect
end

puts solve.call

Benchmark.bm(6) do |x|
  x.report("solve:") { solve.call }
end

File: 39.py
import math
from collections import Counter

perimeters = Counter()

for a in range(1, 500):
    for b in range(1, 500):
        c = math.sqrt(a**2 + b**2)
        p = a + b + c
        if c % 1 != 0 or p > 1000:
            continue
        perimeters[p] += 1

print(perimeters.most_common(1))

